### DISCUSSION TITLE
New problem of C++ modules

### DISCUSSION THREAD

**Discussion by Extra-Creativity** [xmake-io/xmake#4682] [Q&A] [answered]
Visual Studio can compile while xmake cannot, with strange error - redefinition of `std::string_view`.
xmake.lua：
```lua
add_rules("mode.debug", "mode.release")
set_languages("cxxlatest")

target("module-split-impl")
    set_kind("binary")
    add_files("*.cpp")
    add_files("Person.mpp")
```
main.cpp：
```c++
import std;
import Person;

int main()
{
    std::println("---------module-split-impl---------");
    HelloWorld();
    Person person{"Wow", 6};
    person.Print();
    return 0;
}
```
Person.mpp：
```c++
export module Person; // denote that it's a module interface.

import std;

export class Person
{
public:
    Person(const std::string& init_name, std::uint64_t init_id);
    void Print();

private:
    std::string name;
    std::uint64_t id;
};

export void HelloWorld();
```
Peron.cpp：
```c++
module Person; // denote that it's a module implementation
import std;

Person::Person(const std::string& init_name, std::uint64_t init_id):
    name{init_name}, id{init_id} {}

void Person::Print()
{
    std::println("Person #{}: {}", id, name);
}
```
Person-Func.cpp：
```c++
module Person;
import std;

void HelloWorld()
{
    std::println("Hello, world!");
}
```

**Comment by Extra-Creativity** [comment]
I also find that example of module partition in cppreference cannot be compiled in xmake.

**Comment by waruqi** [comment]
@Arthapz Any idea? I have not latest vs version now.

**Comment by Extra-Creativity** [comment]
I'm not sure whether it's related to another VS bug I report [here](https://developercommunity.visualstudio.com/t/C23-module-report-wrong-redefinition/10577433). This example cannot be built either by VS or by xmake.

**Comment by Arthapz** [comment]
well it work on my side

![image](https://github.com/xmake-io/xmake/assets/27437334/d6ee34d9-30c1-4ac8-a09e-707660a17892)

but i'm using VS 17.9.0 Preview 5

**Reply by Extra-Creativity** [reply]
Okay, possibly it's a VS bug. I'd try it in the next VS version. What about the module partition example in cppreference?

**Reply by Arthapz** [reply, marked as answer]
there is an issue with .cpp modules currently
consider using .mpp / .cppm / .ixx for named module / module partition for now
i'll fix that later on
